### ISSUE TITLE
Different numerical results on 1.10 and 1.9.4

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaArrays/StaticArrays.jl#1222]
Hi, 

I just found out that 1.9.4 and 1.10 give different results for a matrix multiply:
```julia

**Comment by mateuszbaran** [comment]
Most likely LLVM generates different SIMD instructions. Is this actually a problem in practice?

**Comment by SimonDanisch** [comment]
Yeah, in Makie this changed the exact float values of markers, which changed their hash in the texture atlas which is shared between Julia versions.
But I'm not sure if it's actually a problem for others or even fixable in StaticArrays.
Just wanted to have a few more eyes in this in case it's an actual bug!

**Comment by mateuszbaran** [comment]
It might be fixable but I would guess that it's not worth the required effort. I really doubt this is actually serious. Lots of people use matrix multiplication from StaticArrays.jl and AFAIK there are no serious issues with it.

**Comment by SimonDanisch** [comment]
Note that Julia's Base matrix multiply doesn't have this issue and that:
```julia
julia> 7.419375817039376e-17 ≈ 2.8053096100683734e-17
false
```
So it could be an issue, I think.

**Comment by mateuszbaran** [comment]
Yeah, Julia's Base uses BLAS which does its own SIMD.

Default `isapprox` is not ideal for comparing two numbers that are supposed to approximate 0 -- for that I usually use absolute tolerance on the order of machine epsilon (`2.220446049250313e-16` for `Float64`, which is more than what we have here).
